crates/sui-core/src/epoch/randomness.rs

```diff
@@ -557,11 +557,32 @@ impl RandomnessManager {
         Ok(())
     }
 
-    /// Reserves the next available round number for randomness generation. Once the given
+    /// Reserves the next available round number for randomness generation if enough time has
+    /// elapsed, or returns None if not yet ready (based on ProtocolConfig setting). Once the given
     /// batch is written, `generate_randomness` must be called to start the process. On restart,
     /// any reserved rounds for which the batch was written will automatically be resumed.
-    pub fn reserve_next_randomness(&mut self, batch: &mut DBBatch) -> SuiResult<RandomnessRound> {
-        let tables = self.tables()?;
+    pub fn reserve_next_randomness(
+        &mut self,
+        commit_timestamp: TimestampMs,
+        batch: &mut DBBatch,
+    ) -> SuiResult<Option<RandomnessRound>> {
+        let epoch_store = self.epoch_store()?;
+        let tables = epoch_store.tables()?;
+
+        let last_round_timestamp = tables
+            .randomness_last_round_timestamp
+            .get(&SINGLETON_KEY)
+            .expect("typed_store should not fail");
+        if let Some(last_round_timestamp) = last_round_timestamp {
+            if commit_timestamp - last_round_timestamp
+                < self
+                    .epoch_store()?
```

Comment:
you already loaded epoch_store earlier

Reply:
thanks for catching, I added that after writing this line, my bad!